src/Nethermind/Nethermind.Core.Test/Encoding/ConsensusRequestDecoderTests.cs

```diff
@@ -0,0 +1,126 @@
+using Nethermind.Serialization.Rlp;
+using NUnit.Framework;
+using Nethermind.Core.Test.Builders;
+
+namespace Nethermind.Core.Test.Encoding;
+
+public class ConsensusRequestDecoderTests
+{
+    [Test]
+    public void Should_decode_deposit()
+    {
+        ConsensusRequest deposit = new Deposit()
+        {
+            Index = long.MaxValue,
+            PubKey = KeccakTests.KeccakOfAnEmptyString.ToBytes(),
+            Signature = KeccakTests.KeccakOfAnEmptyString.ToBytes(),
+            WithdrawalCredentials = KeccakTests.KeccakOfAnEmptyString.ToBytes(),
+            Amount = int.MaxValue
+        };
+
+        byte[] rlp = Rlp.Encode(deposit).Bytes;
+        ConsensusRequest decoded = Rlp.Decode<ConsensusRequest>(rlp);
+
+        decoded.Should().BeEquivalentTo(deposit);
+    }
+
+    [Test]
+    public void Should_decode_withdrawalRequest()
+    {
+        byte[] validatorPubkey = new byte[48];
+        validatorPubkey[11] = 11;
+        ConsensusRequest withdrawalRequest = new WithdrawalRequest()
+        {
+            SourceAddress = TestItem.AddressA,
+            ValidatorPubkey = validatorPubkey,
+            Amount = int.MaxValue
+        };
+
+        byte[] rlp = Rlp.Encode(withdrawalRequest).Bytes;
+        ConsensusRequest decoded = Rlp.Decode<ConsensusRequest>(rlp);
+
+        decoded.Should().BeEquivalentTo(withdrawalRequest);
+    }
+
+    [Test]
+    public void Should_decode_deposit_with_ValueDecoderContext()
+    {
+        ConsensusRequest deposit = new Deposit()
+        {
+            Index = long.MaxValue,
+            PubKey = KeccakTests.KeccakOfAnEmptyString.ToBytes(),
+            Signature = KeccakTests.KeccakOfAnEmptyString.ToBytes(),
+            WithdrawalCredentials = KeccakTests.KeccakOfAnEmptyString.ToBytes(),
+            Amount = int.MaxValue
+        };
+        RlpStream stream = new(1024);
+        ConsensusRequestDecoder codec = new();
+
+        codec.Encode(stream, deposit);
+
+        Rlp.ValueDecoderContext decoderContext = new(stream.Data.AsSpan());
+        Deposit? decoded = (Deposit?)codec.Decode(ref decoderContext);
+
+        decoded.Should().BeEquivalentTo(deposit);
+    }
+
+    [Test]
+    public void Should_decode_withdrawalRequest_with_ValueDecoderContext()
+    {
+        ConsensusRequest withdrawalRequest = new WithdrawalRequest()
+        {
+            SourceAddress = TestItem.AddressA,
+            ValidatorPubkey = KeccakTests.KeccakOfAnEmptyString.ToBytes(),
+            Amount = int.MaxValue
+        };
+        RlpStream stream = new(1024);
+        ConsensusRequestDecoder codec = new();
+
+        codec.Encode(stream, withdrawalRequest);
+
+        Rlp.ValueDecoderContext decoderContext = new(stream.Data.AsSpan());
+        WithdrawalRequest? decoded = (WithdrawalRequest?)codec.Decode(ref decoderContext);
+
+        decoded.Should().BeEquivalentTo(withdrawalRequest);
+    }
+
+    [Test]
+    public void Should_encode_deposit_same_for_Rlp_Encode_and_DepositDecoder_Encode()
```

Comment:
we need tests for decoders many different types and for array of consensus requests